### PR TITLE
Cleanup doc_fragments/aws.py

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Allow to pull the requirements from `aws_boto3.py` doc fragments instead of from `aws.py`.
This also allows collections importing docs fragments from amazon.aws to define their own boto3, botocore and python requirements.

##### ISSUE TYPE

- Docs Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by tremble** [comment]
Note: This would technically be a breaking change, and as such shouldn't be backported.

**Review by tremble — CHANGES_REQUESTED** [review]
Generally in favour of the change, however the developer docs also need updated:

https://github.com/ansible-collections/amazon.aws/blob/main/docs/docsite/rst/dev_guidelines.rst#common-documentation-fragments-for-connection-parameters

**Comment by alinabuzachis** [comment]
> Generally in favour of the change, however the developer docs also need updated:
> 
> https://github.com/ansible-collections/amazon.aws/blob/main/docs/docsite/rst/dev_guidelines.rst#common-documentation-fragments-for-connection-parameters

@tremble done, thanks!

**Comment by alinabuzachis** [comment]
recheck

**Review by jillr — APPROVED** [review]

**Review by tremble — APPROVED** [review]

File: docs/docsite/rst/dev_guidelines.rst [hunk 1]
```diff
@@ -299,6 +299,7 @@ that should be included into almost all AWS modules:
```
Comment (gravesm) [inline docs/docsite/rst/dev_guidelines.rst:299]: Github won't let me put this comment on a line you didn't change, but two lines above this "There are two" should probably be updated to say "There are four".

**Review by goneri — APPROVED** [review]

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by jillr** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by tremble** [comment]
Will probably still fail.  The inventory tests take about 45 minutes.

This might help:
https://github.com/ansible-collections/amazon.aws/pull/1070/files#diff-797630fda5cc98086f92121dd44832753fb831ff7ef40616b1ac8f2436f03f65

**Comment by tremble** [comment]
recheck

**Comment by tremble** [comment]
recheck

**Comment by tremble** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Pull request merged**